MODULE.bazel

```diff
@@ -0,0 +1,174 @@
+bazel_dep(name = "grpc", version = "1.56.3.bcr.1", repo_name = "com_github_grpc_grpc")
+bazel_dep(name = "protobuf", version = "23.1", repo_name = "com_google_protobuf")
+bazel_dep(name = "rules_cc", version = "0.0.9")
+bazel_dep(name = "rules_go", version = "0.46.0", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "rules_jvm_external", version = "6.0")
+bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
+
+non_module_deps = use_extension("//:repositories.bzl", "grpc_java_repositories_extension")
+use_repo(
+    non_module_deps,
+    "com_github_cncf_xds",
+    "envoy_api",
+    "io_grpc_grpc_proto",
+)
+
+grpc_repo_deps_ext = use_extension("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_repo_deps_ext")
+use_repo(
+    grpc_repo_deps_ext,
+    "com_envoyproxy_protoc_gen_validate",
+    "com_github_cncf_udpa",
+    "opencensus_proto",
+)
+
+maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
+maven.install(
+    artifacts = [
+        "com.google.android:annotations:4.1.1.4",
+        "com.google.api.grpc:proto-google-common-protos:2.29.0",
+        "com.google.auth:google-auth-library-credentials:1.22.0",
+        "com.google.auth:google-auth-library-oauth2-http:1.22.0",
+        "com.google.auto.value:auto-value-annotations:1.10.4",
+        "com.google.auto.value:auto-value:1.10.4",
+        "com.google.code.findbugs:jsr305:3.0.2",
+        "com.google.code.gson:gson:2.10.1",
+        "com.google.errorprone:error_prone_annotations:2.23.0",
+        "com.google.guava:failureaccess:1.0.1",
+        "com.google.guava:guava:32.1.3-android",
+        "com.google.re2j:re2j:1.7",
+        "com.google.truth:truth:1.1.5",
+        "com.squareup.okhttp:okhttp:2.7.5",
+        "com.squareup.okio:okio:2.10.0",  # 3.0+ needs swapping to -jvm; need work to avoid flag-day
+        "io.netty:netty-buffer:4.1.100.Final",
+        "io.netty:netty-codec-http2:4.1.100.Final",
+        "io.netty:netty-codec-http:4.1.100.Final",
+        "io.netty:netty-codec-socks:4.1.100.Final",
+        "io.netty:netty-codec:4.1.100.Final",
+        "io.netty:netty-common:4.1.100.Final",
+        "io.netty:netty-handler-proxy:4.1.100.Final",
+        "io.netty:netty-handler:4.1.100.Final",
+        "io.netty:netty-resolver:4.1.100.Final",
+        "io.netty:netty-tcnative-boringssl-static:2.0.61.Final",
+        "io.netty:netty-tcnative-classes:2.0.61.Final",
+        "io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.100.Final",
+        "io.netty:netty-transport-native-unix-common:4.1.100.Final",
+        "io.netty:netty-transport:4.1.100.Final",
+        "io.opencensus:opencensus-api:0.31.0",
+        "io.opencensus:opencensus-contrib-grpc-metrics:0.31.0",
+        "io.perfmark:perfmark-api:0.26.0",
+        "junit:junit:4.13.2",
+        "org.apache.tomcat:annotations-api:6.0.53",
+        "org.codehaus.mojo:animal-sniffer-annotations:1.23",
+    ],
+    fail_if_repin_required = True,
+    lock_file = "//:maven_install.json",
```

Comment:
Remove the lock file. Only one module can provide the lock file. We want our consumers to provide it.

Reply:
the recommendation [here](https://bazelbuild.slack.com/archives/CDCE4DFAP/p1711743105245039?thread_ts=1711742750.217869&cid=CDCE4DFAP) from the rules_jvm_external maintainers was to always have a lockfile. although I think from that discussion it's more about how downstream consumers wouldn't need to do resolution in that case, which might never work with a library like this where presumably some java dep resolution will happen downstream.

Reply:
removed!